File: isPalindrome/palindromeMethods.py
# Methods for determining if a string is or contains a palindrome

def isPalindrome(s):
    # Determines if string input is a palindrome or not
    # One character strings will be considered palindromes
    for i in range(len(s) - 1):
        # loops through string indices
        if s[i] != s[(-1) - i]:
            # checks if 1st character of string matches last character
            return False
    return True


##print(isPalindrome('a'))
##print(isPalindrome('aba'))
##print(isPalindrome('abad'))
##print(isPalindrome('abcba'))
##print(isPalindrome('abcde'))

def containsPalindrome(s):
    # Determines if string input contains a palindrome larger than
    # an individual character
    if len(s) <= 1:
        # Base case: If the string is a character, return False
        return False
    elif isPalindrome(s) == True:
        # Check for if string is a palindrome
        return True
    else:
        # Recursive call
        return containsPalindrome(s[1:]) or containsPalindrome(s[0:-1])

##print(containsPalindrome('a'))
##print(containsPalindrome('abc'))
##print(containsPalindrome('abcba'))
##print(containsPalindrome('alab'))
##print(containsPalindrome('abab'))
##print(containsPalindrome('abcb'))

File: isPalindrome/PalindromeMethods.java
package isPalindrome;

//Method for determining if a string is or contains a palindrome
public class PalindromeMethods {
	public static Boolean isPalindrome(String s)
	{
		int length = s.length();
		for(int i = 0; i < length/2; i++)
		{
			if(s.charAt(i) != s.charAt(length-i-1))
				return false;
		}
		return true;
	}
	
	public static Boolean containsPalindrome(String s)
	{
		int length = s.length();
		if(length <= 1)
			return false;
		if(isPalindrome(s) == true)
			return true;
		else
			return containsPalindrome(s.substring(1, s.length())) || containsPalindrome(s.substring(0,s.length()-1));
	}
	
	public static void main(String[] args) {
		System.out.println(isPalindrome("a")); //true
		System.out.println(isPalindrome("aba")); //true
		System.out.println(isPalindrome("abad")); //false
		System.out.println(isPalindrome("abcba")); //true
		System.out.println(isPalindrome("abcde")); //false
		System.out.println(containsPalindrome("a")); //false
		System.out.println(containsPalindrome("abc")); //false
		System.out.println(containsPalindrome("abcba")); //true
		System.out.println(containsPalindrome("alab")); //true
		System.out.println(containsPalindrome("abcb")); //true
	}
}